### DISCUSSION TITLE
oh-my-posh V3 over SSH (or even just on VIM exit) causes strange colors in terminal

### DISCUSSION THREAD

**Discussion by mikeev261** [JanDeDobbeleer/oh-my-posh#472] [Troubleshoot] [answered]
Using oh-my-posh V3 between all Win10 machines (all with updated versions of OMP V3) in Powershell Core 7 can cause the colors to render incorrectly. I'm using a [customized theme](https://pastebin.com/i6kWL21K) (based on the default JanDeDobbeleer theme), but I don't think that's what's causing the colors to change in general. 

Here's what I'm observing (with some examples): 

Windows Terminal + Powershell Core 7 + OMP, just navigating = correct colors

![image](https://user-images.githubusercontent.com/4935661/109502559-2374a800-7a67-11eb-9004-15c8a82b906f.png)

VsCode Terminal + Powershell Core 7 + OMP, just navigating = correct colors 

![image](https://user-images.githubusercontent.com/4935661/109502274-c678f200-7a66-11eb-83b1-faad453ba686.png)

VsCode Terminal + Powershell Core 7 + OMP, after running VIM =  old terminal colors go bad after VIM exit. Fresh terminal lines are fine. 

![image](https://user-images.githubusercontent.com/4935661/109502420-f1fbdc80-7a66-11eb-9df5-b8324b50e401.png)

Windows Terminal + Powershell Core 7 + OMP, after running VIM = old terminal colors go bad after VIM exit. Fresh terminal lines are fine. 

Windows Terminal + Powershell Core 7 + OMP + SSH = correct colors

Windows Terminal + Powershell Core 7 + OMP  + SSH, after running VIM = old terminal colors go bad after VIM exit. Fresh terminal lines are fine. 

![image](https://user-images.githubusercontent.com/4935661/109501788-17d4b180-7a66-11eb-85bd-456d0ab93afc.png)

VsCode Terminal + Powershell Core 7 + OMP + SSH = **always** bad colors

![image](https://user-images.githubusercontent.com/4935661/109502118-8fa2dc00-7a66-11eb-85a4-ddc993282335.png)

Not sure if this is an OMP V3 bug or not, but if it isn't, is there a way to fix it?

**Comment by JanDeDobbeleer** [comment, marked as answer]
@mikeev261 this is indeed caused by Windows Terminal and/or VIM. There's already a [discussion](https://github.com/JanDeDobbeleer/oh-my-posh/discussions/229#discussioncomment-185682) about this.

To be clear, this only happens inside the Windows console hosts. The standard Powershell progress bar (not the new preview one) has the same effect of ruining the colours. So something that needs fixing there.

**Reply by mikeev261** [reply]
OK, thanks- sorry for the duplicate.

**Reply by JanDeDobbeleer** [reply]
No worries. Thanks for logging this elaborately, makes my life easier 👍🏻
